js_hw6/task_02.js

```diff
@@ -0,0 +1,267 @@
+// Task 2. Перед вами структура компани, и ниже представлены задания, относящиеся к ней. 
+// В заданиях по максимуму использовать методы массивов, создавать функции-помощники, выполняющие дополнительные действия,
+// чтобы ваши функции выполняли строго одну работу. ЭТО ОЧЕНЬ ВАЖНО!
+
+const enterprises = [
+  {
+    id: 1,
+    name: "Предприятие 1",
+    departments: [
+      {
+        id: 2,
+        name: "Отдел тестирования",
+        employees_count: 10,
+      },
+      {
+        id: 3,
+        name: "Отдел маркетинга",
+        employees_count: 20,
+      },
+      {
+        id: 4,
+        name: "Администрация",
+        employees_count: 15,
+      },
+    ]
+  },
+  {
+    id: 5,
+    name: "Предприятие 2",
+    departments: [
+      {
+        id: 6,
+        name: "Отдел разработки",
+        employees_count: 50,
+      },
+      {
+        id: 7,
+        name: "Отдел маркетинга",
+        employees_count: 20,
+      },
+      {
+        id: 8,
+        name: "Отдел охраны труда",
+        employees_count: 5,
+      },
+    ]
+  },
+  {
+    id: 9,
+    name: "Предприятие 3",
+    departments: [
+      {
+        id: 10,
+        name: "Отдел аналитики",
+        employees_count: 0,
+      },
+    ]
+  }
+]
+
+// Задания:
+// 1. Вывести все предприятия и их отделы. Рядом указать количество сотрудников. Для предприятия посчитать сумму всех сотрудников во всех отделах.
+
+// **Пример:**
+
+// Предприятие 1 (45 сотрудников)
+// - Отдел тестирования (10 сотрудников)
+// - Отдел маркетинга (20 сотрудников)
+// - Администрация (15 человек)
+// Предприятие 2 (75 сотрудников)
+// - Отдел разработки (50 сотрудников)
+// - Отдел маркетинга (20 сотрудников)
+// - Отдел охраны труда (5 сотрудников)
+// Предприятие 3 (нет сотрудников)
+// - Отдел аналитики (нет сотрудников)
+
+function logEmployess (enterprises) {
+     for (const obj in enterprises) {
+        console.log(`${enterprises[obj].name} (${getAllEmployeesInEnterprise(enterprises[obj].departments)} сотрудников)`);
+        enterprises[obj].departments.forEach((department) => {
+            console.log(`- ${department.name} (${(department.employees_count) === 0 ? 'нет' : department.employees_count} сотрудников)`);
+        });
+     };
+    };
+
+
+function getAllEmployeesInEnterprise (departments) {
+    countAll = 0;
+    for (let dep of departments) {
+        countAll += dep.employees_count;
+    }
+    return (countAll === 0) ? 'нет' : countAll;
+};
+
+logEmployess(enterprises);
+
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+function getEntrepriseName (depIdOrName) {
```

Comment:
For this task you can implement full logic using only this function, but further you'll need the helper functions